docs/docs/store.mdx

```diff
@@ -37,48 +37,134 @@ When you want to update the store, you can call the `update()` method passing th
 
 ```ts title="session.service.ts"
 import { SessionStore } from './session.store';
- 
+
 export class SessionService {
   constructor(private sessionStore: SessionStore) {}
 
   updateUserName(newName: string) {
     this.sessionStore.update({ name: newName });
-  }  
+  }
 }
 ```
 
 The second `update()` option gives you more control. It receives a `callback` function, which gets the current state, and returns a new **immutable** state, which will be the new value of the store. For example:
 
 ```ts title="session.service.ts"
 import { SessionStore } from './session.store';
- 
+
 export class SessionService {
   constructor(private sessionStore: SessionStore) {}
 
   updateUserName(newName: string) {
     this.sessionStore.update(state => ({
       name: newName
     }));
-  }  
+  }
+}
+```
+
+### `overwrite()`
+
+When you don't just want to update the current store state but replace the state completely, you can call the `overwrite()` method passing the new `state`:
+
+```ts title="session.service.ts"
+import { SessionStore } from './session.store';
+
+export class SessionService {
+  constructor(private sessionStore: SessionStore) {}
+
+  updateUserName(newName: string) {
+    this.sessionStore.overwrite({ name: newName });
+  }
+}
+```
+
+As with `update()` the `overwrite()` method also has a second option which gives you more control. It receives a `callback` function, which gets the current state, and returns a new **immutable** state, which will be the new value of the store. For example:
+
+```ts title="session.service.ts"
+import { SessionStore } from './session.store';
+
+export class SessionService {
+  constructor(private sessionStore: SessionStore) {}
+
+  updateUserName(newName: string) {
+    this.sessionStore.overwrite(state => ({
+      name: newName
+    }));
+  }
 }
 ```
 
+You should use `overwrite()` over `update()` when you want to completely replace the current state at the top level.
+
+```ts title="session.store.ts"
+type SessionState = SessionStateAuthenticated | SessionStateUnauthenticated;
+
+interface SessionStateAuthenticated {
+  _tag: 'session-state-authenticated';
+  user: User;
+}
+
+interface SessionStateUnauthenticated {
+  _tag: 'session-state-unauthenticated'
+}
+
+interface User {
+  uid: string;
+  displayName: string;
+}
+
+```
+
+```json "current state"
+{
+  "_tag": "session-state-authenticated",
+  "user": {
+    "uid": "123",
+    "displayName": "John Doe"
+  }
+}
+```
+
+Using the above state model and value of current state, if we call `this.store.update({ _tag: 'session-state-unauthenticated' })` our state will be updated to the following:
```

Comment:
I don't think we need all of this. The method name is straightforward.

Reply:
I was concerned that newer users may be unsure when to use each method, but it could be way to verbose.

Reply:
The `overwrite` name is clear enough.

Reply:
I'll make the change.